lib/Exporter/Writer/KnpSnappyWriter.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+/*
+ * This file is part of the Sonata package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Exporter\Writer;
+
+use Knp\Snappy\Pdf;
+
+/**
+ * Export data in Pdf format.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class KnpSnappyWriter implements WriterInterface
+{
+    /**
+     * @var string
+     */
+    protected $filename;
+
+    /**
+     * @var string
+     */
+    protected $html;
+
+    /**
+     * @var bool
+     */
+    protected $showHeaders;
+
+    /**
+     * @var bool
+     */
+    protected $showBorders;
+
+    /**
+     * @var array
+     */
+    protected $snappyOptions;
+
+    /**
+     * @var string
+     */
+    protected $wktmltopdfBinary;
+
+    /**
+     * @var int
+     */
+    protected $position;
+
+    /**
+     * @var Pdf
+     */
+    protected $snappy;
+
+    /**
+     * @throws \RuntimeException
+     *
+     * @param string $filename
+     * @param bool   $showHeaders
+     * @param bool   $showBorders
+     * @param array  $snappyOptions
+     * @param string $wkhtmltopdfBinary Path of the wkhtmltopdf binary
+     */
+    public function __construct($filename, $showHeaders = true, $showBorders = true, $snappyOptions = null, $wkhtmltopdfBinary = null)
+    {
+        $this->filename          = $filename;
+        $this->showHeaders       = $showHeaders;
+        $this->showBorders       = $showBorders;
+        $this->snappyOptions     = $snappyOptions;
+        $this->wkhtmltopdfBinary = $wkhtmltopdfBinary;
+        $this->position          = 0;
+        $this->snappy            = new Pdf();
+
+        if (is_file($filename)) {
+            throw new \RuntimeException(sprintf('The file %s already exist', $filename));
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function open()
+    {
+        $this->html = '<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><meta name=ProgId content=Pdf.Document><meta name=Generator content="https://github.com/sonata-project/exporter"></head><body><table';
```

Comment:
Can you create a tmpfile for the html content ? This to avoid extra memory usage.